ladybug/windrose.py

```diff
@@ -188,19 +232,37 @@ def legend_parameters(self):
         Default segment count is 10.
         """
         if self._legend_parameters is None:
-            self._legend_parameters = LegendParameters(segment_count=10)
+            self._legend_parameters = self.container.legend_parameters
         return self._legend_parameters
 
     @legend_parameters.setter
     def legend_parameters(self, legend_parameters):
         assert isinstance(legend_parameters, LegendParameters), 'legend_parameters' \
             ' must be a LegendParameters. Got {}.'.format(type(legend_parameters))
-
-        # This will mutate the legend_parameters in the windrose Legend object.
         self._compass = None
         self._container = None
         self._legend_parameters = legend_parameters
 
+    @property
+    def north(self):
+        """North orientation for windrose by degrees in counterclockwise orienation.
+
+        This must be a number between -360 and 360. 0 is North, 90 is West and 270
+        is East.
+        """
+        if self._north is None:
+            self._north = self.DEFAULT_NORTH
```

Comment:
Why not just `return self.DEFAULT_NORTH`? Here you are loosing track of the fact that the user did not specify this value for `_north`. Or, if this information about whether the north was user-assigned is not important, why bother setting `_north` to `None` in the first place.

Reply:
I see your point, it doesn't have to be set to None in the first place. I'll set it to the default value in the __init__ function.